From: David Herrmann <dh.herrmann@gmail.com>
Date: Wed, 2 Apr 2014 16:20:13 +0200
Subject: logind: don't print error if devices vanish during ACL-init

If a device is unplugged while we initialize it, we will get ENOENT for
ACL-init (and related stuff). We currently print errors then, which is
misleading. Print a debug-message early and continue.

(cherry picked from commit 8016b904849a2d2bd8b0171ed8493db5524b5ff7)
---
 src/login/logind-acl.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/login/logind-acl.c b/src/login/logind-acl.c
index dc86f0f1..4bbeb64e 100644
--- a/src/login/logind-acl.c
+++ b/src/login/logind-acl.c
@@ -279,7 +279,9 @@ int devnode_acl_all(struct udev *udev,
 
                 log_debug("Fixing up ACLs at %s for seat %s", n, seat);
                 k = devnode_acl(n, flush, del, old_uid, add, new_uid);
-                if (k < 0)
+                if (k == -ENOENT)
+                        log_debug("Device %s disappeared while setting ACLs", n);
+                else if (k < 0)
                         r = k;
         }
 
